src/merge_tools.rs

```diff
@@ -227,7 +227,7 @@ pub fn run_mergetool(
                 // TODO: Should actually ignore the error here, or have a warning.
                 set_readonly_recursively(&path).map_err(ExternalToolError::SetUpDirError)?;
             }
-            Ok((*role, path))
+            Ok((*role, path.into_os_string().into_string().unwrap()))
```

Comment:
Maybe change `unwrap` to `expect("We are assuming the temporary dir is valid utf-8")` for documentation purposes and in the very unlikely case somebody sees this panic.